client/wc-api/settings/operations.js

```diff
@@ -20,57 +19,61 @@ function update( resourceNames, data, fetch = apiFetch ) {
 }
 
 function readSettings( resourceNames, fetch ) {
-	if ( resourceNames.includes( 'settings' ) ) {
-		const url = NAMESPACE + '/settings/wc_admin';
+	const filteredNames = resourceNames.filter( name => {
+		return name.startsWith( 'settings/' );
+	} );
 
-		return [
-			fetch( { path: url } )
-				.then( settingsToSettingsResource )
-				.catch( error => {
-					return { [ 'settings' ]: { error: String( error.message ) } };
-				} ),
-		];
-	}
-	return [];
+	return filteredNames.map( async resourceName => {
+		const url = NAMESPACE + '/' + resourceName;
+
+		return fetch( { path: url } )
+			.then( settingsToSettingsResource.bind( null, resourceName ) )
+			.catch( error => {
+				return { [ resourceName ]: { error: String( error.message ) } };
+			} );
+	} );
 }
 
 function updateSettings( resourceNames, data, fetch ) {
-	const resourceName = 'settings';
-	const settingsFields = [
-		'woocommerce_excluded_report_order_statuses',
-		'woocommerce_actionable_order_statuses',
-	];
+	const filteredNames = resourceNames.filter( name => {
+		return name.startsWith( 'settings/' );
+	} );
 
-	if ( resourceNames.includes( resourceName ) ) {
-		const url = NAMESPACE + '/settings/wc_admin/';
-		const settingsData = pick( data[ resourceName ], settingsFields );
-
-		const promises = Object.keys( settingsData ).map( setting => {
-			return fetch( {
-				path: url + setting,
-				method: 'POST',
-				data: { value: settingsData[ setting ] },
-			} )
-				.then( settingToSettingsResource.bind( null, data.settings ) )
-				.catch( error => {
-					return { [ resourceName ]: { error } };
-				} );
+	return filteredNames.map( async resourceName => {
+		const url = NAMESPACE + '/' + resourceName + '/batch';
+		const settingsData = Object.keys( data[ resourceName ] ).map( key => {
+			return { id: key, value: data[ resourceName ][ key ] };
 		} );
 
-		return promises;
-	}
-	return [];
+		return fetch( {
+			path: url,
+			method: 'POST',
+			data: { update: settingsData },
+		} )
+			.then( settingToSettingsResource.bind( null, resourceName ) )
+			.catch( error => {
+				return { [ resourceName ]: { error } };
+			} );
+	} );
 }
 
-function settingsToSettingsResource( settings ) {
+function settingsToSettingsResource( resourceName, settings ) {
 	const settingsData = {};
 	settings.forEach( setting => ( settingsData[ setting.id ] = setting.value ) );
-	return { [ 'settings' ]: { data: settingsData } };
+	return { [ resourceName ]: { data: settingsData } };
 }
 
-function settingToSettingsResource( settings, setting ) {
-	settings[ setting.id ] = setting.value;
-	return { [ 'settings' ]: { data: settings } };
+function settingToSettingsResource( resourceName, data ) {
+	const settings = {};
+	if ( 'undefined' === typeof data.update ) {
+		return '';
+	}
+
+	// @todo This will only return updated fields so fields
+	// not updated may be temporarily overwritten in the store.
+	data.update.forEach( setting => ( settings[ setting.id ] = setting.value ) );
```

Comment:
While this doesn't affect wc-admin currently, it doesn't feel right to overwrite the existing store with partial data.

I'm not aware of any way to grab existing data from here and merge, but that would be ideal.  Open to other thoughts here or following up in another PR.

Reply:
Hmm, I don't know of a way to do that here either. Maybe @coderkevin has some ideas?